server/datastore/mysql/migrations/tables/20240607133721_ReconcileSoftwareTitles.go

```diff
@@ -0,0 +1,53 @@
+package tables
+
+import (
+	"database/sql"
+	"fmt"
+)
+
+func init() {
+	MigrationClient.AddMigration(Up_20240607133721, Down_20240607133721)
+}
+
+func Up_20240607133721(tx *sql.Tx) error {
+
+	// For users that are not running vulnerabilities job, we need to ensure that software_titles are up-to-date
+
+	upsertTitlesStmt := `
+INSERT INTO software_titles (name, source, browser)
+SELECT DISTINCT
+	name,
+	source,
+	browser
+FROM
+	software s
+WHERE
+	NOT EXISTS (SELECT 1 FROM software_titles st WHERE (s.name, s.source, s.browser) = (st.name, st.source, st.browser))
+ON DUPLICATE KEY UPDATE software_titles.id = software_titles.id`
```

Comment:
This is how it was done in `ReconcileSoftwareTitles` method, so I kept it consistent. I'm guessing it was done so that the INSERT statement doesn't fail. It does seem like `INSERT IGNORE` would be less wordy/faster.